### PR TITLE
Explicitly type getDetectorInfo and remove duplicate code

### PR DESCRIPTION
**Pull request by peterbom**

### CODE REVIEW

File: src/tree/aksClusterTreeItem.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export interface AksClusterTreeNode {
 export default class AksClusterTreeItem extends AzureTreeItem implements AksClusterTreeNode {
     constructor(
         parent: AzExtParentTreeItem,
```
Comment (Tatsinnit) [inline src/tree/aksClusterTreeItem.ts:20]: ☕️ let’s discuss about this change so I think by contract the “cloud explorer” guarantee target will always have name and type, hence let’s relive this small change from this and and the if conditional check. Unless you were able to replicate something. 

We could make this property not private in separate Pr and involve Ivan as well. What do you think? Just throwing some ideas.

Comment (Tatsinnit) [reply src/tree/aksClusterTreeItem.ts:20]: ❤️❤️❤️ cool, let’s drop this minor change and I shall merge this good one into the feature branch.

thanks heaps , this is soo nice ❤️😊🙏

File: src/tree/aksClusterTreeItem.ts [hunk 2]
```diff
@@ -36,7 +36,11 @@ export default class AksClusterTreeItem extends AzureTreeItem implements AksClus
     }
 
     public get name(): string {
-        return this.resource.name || '';
+        return this.resource.name!;
```
Comment (peterbom) [inline src/tree/aksClusterTreeItem.ts:39]: Note: I changed this because that's my preferred approach for the `resourceType` property I'm adding, and I think they should be consistent.

If both `name` and `type` on the resource really should always be defined (in this context at least), then I think it makes sense for the code to throw rather than return empty string (which would fail in weird ways in other places).

On the other hand, if they might not always be defined, then we should change these properties to return `string | undefined` and let consumers deal with it.
